tests/shared/utils.spec.js

```diff
@@ -124,4 +127,191 @@ describe('Shared: Utils', function() {
 
   });
 
+  describe('hideObjectAbsolutePaths', function() {
+
+    it('should return undefined if given undefined', function() {
+      m.chai.expect(utils.hideObjectAbsolutePaths(undefined)).to.be.undefined;
+    });
+
+    it('should return null if given null', function() {
+      m.chai.expect(utils.hideObjectAbsolutePaths(null)).to.be.null;
+    });
+
+    it('should return a clone of the object if there are no paths in the object', function() {
+      const object = {
+        numberProperty: 1,
+        nested: {
+          otherProperty: 'value'
+        }
+      };
+      m.chai.expect(utils.hideObjectAbsolutePaths(object)).to.not.equal(object);
+      m.chai.expect(utils.hideObjectAbsolutePaths(object)).to.deep.equal(object);
+    });
+
+    describe('given UNIX paths', function() {
+
+      beforeEach(function() {
+        this.isAbsolute = path.isAbsolute;
+        this.basename = path.basename;
+        this.platform = os.platform;
+        path.isAbsolute = path.posix.isAbsolute;
+        path.basename = path.posix.basename;
+        os.platform = _.constant('linux');
+      });
+
+      afterEach(function() {
+        path.isAbsolute = this.isAbsolute;
+        path.basename = this.basename;
+        os.platform = this.platform;
+      });
+
+      it('should replace absolute paths with the basename', function() {
+        const object = {
+          prop1: 'some value',
+          prop2: '/home/john/rpi.img'
+        };
+
+        m.chai.expect(utils.hideObjectAbsolutePaths(object)).to.deep.equal({
+          prop1: 'some value',
+          prop2: 'rpi.img'
+        });
+
+      });
+
+      it('should replace nested absolute paths with the basename', function() {
+        const object = {
+          nested: {
+            path: '/home/john/rpi.img'
+          }
+        };
+
+        m.chai.expect(utils.hideObjectAbsolutePaths(object)).to.deep.equal({
+          nested: {
+            path: 'rpi.img'
+          }
+        });
+      });
+
+      it('should not alter /dev/sdb', function() {
+        const object = {
+          nested: {
+            path: '/dev/sdb'
+          }
+        };
+
+        m.chai.expect(utils.hideObjectAbsolutePaths(object)).to.deep.equal({
+          nested: {
+            path: '/dev/sdb'
+          }
+        });
+      });
+
+      it('should handle arrays', function() {
+        m.chai.expect(utils.hideObjectAbsolutePaths({
+          foo: 'foo',
+          bar: [ {
```

Comment:
Can we put the curly braces in separate lines, for consistency purposes? For example:

```js
foo: 'foo',
bar: [
  {
    path: '/foo/bar/baz'
  },
  {
    path: '/foo/bar/baz'
  },
  {
    path: '/foo/bar/baz'
  }
]
```